index.php

```diff
@@ -47,80 +58,344 @@
+                }
+                else if(strtolower($result['email']) == strtolower($email)) {
+                    throw new RuntimeException("Email address already in use!");
+                }
+            }
+
+            // Gender
+            if(!in_array($gender, User::$genders, true)) {
+                throw new RuntimeException("Invalid gender!");
+            }
+
+            // Village
+            if(!isset($villages[$village])) {
+                throw new RuntimeException("Invalid village!");
+            }
+
+            // Encrypt password
+            $password = $system->hash_password($password);
+
+            $verification_code = sha1(mt_rand(1, 1337000));
+
+            User::create(
+                $system,
+                $user_name,
+                $password,
+                $email,
+                $gender,
+                $village,
+                $villages[$village]['location'],
+                $verification_code
+            );
+
+            $subject = 'Shinobi-Chronicles account verification';
+            $message = "Welcome to Shinobi-Chronicles RPG. Please visit the link below to verify your account: \r\n
+		    {$system->router->base_url}register.php?act=verify&username={$user_name}&verify_key=$verification_code";
+            $headers = "From: Shinobi-Chronicles<" . System::SC_ADMIN_EMAIL . ">" . "\r\n";
+            $headers .= "Reply-To: " . System::SC_NO_REPLY_EMAIL . "\r\n";
+            if(mail($email, $subject, $message, $headers)) {
+                ;
+                $system->message("Account created!<br />Please check the email that you registered with for the verification  link (Be sure to check your spam folder as well)!");
+                $login_message_text = "Account created!<br />Please check the email that you registered with for the verification  link (Be sure to check your spam folder as well)!";
+            }
+            else {
+                $system->message("There was a problem sending the email to the address provided: $email Please contact a staff member on the forums for manual activation.");
+                $login_message_text = "There was a problem sending the email to the address provided: $email Please contact a staff member on the forums for manual activation.";
+            }
+        } catch (Exception $e) {
+            $system->db->rollbackTransaction();
+            $system->message($e->getMessage());
+            error_log($e->getMessage());
+            $register_error_text = $e->getMessage();
+
+            $register_pre_fill['user_name'] = isset($_POST['user_name']) ? $_POST['user_name'] : "";
+            $register_pre_fill['email'] = isset($_POST['email']) ? $_POST['email'] : "";
+            $register_pre_fill['gender'] = isset($_POST['gender']) ? $_POST['gender'] : "";
+            $register_pre_fill['village'] = isset($_POST['village']) ? $_POST['village'] : "";
+        }
+    }
```

Comment:
I want to extract these huge blocks of logic out so `index` can be lighter-weight, but I'll take that as a follow-up for me to do